uefi-test-runner/build.py

```diff
@@ -132,7 +130,29 @@ def run_qemu(headless):
     if VERBOSE:
         print(' '.join(cmd))
 
-    sp.run(cmd).check_returncode()
+    # This regex can be used to detect and strip ANSI escape codes when
+    # analyzing the output of the test runner.
+    ansi_escape = re.compile(r'(\x9B|\x1B\[)[0-?]*[ -/]*[@-~]')
+
+    # Start QEMU
+    qemu = sp.Popen(cmd, stdout=sp.PIPE, universal_newlines=True)
+
+    # Iterate over stdout...
+    for line in qemu.stdout:
+        # Strip ending and trailing whitespace + ANSI escape codes for analysis
+        stripped = ansi_escape.sub('', line.strip())
+
+        # Skip empty lines
+        if not stripped:
+            continue
+
+        # Print out the processed QEMU output to allow logging & inspection
+        print(stripped)
```

Comment:
You won't believe how clean the logs are now! Just check out Travis :)

Reply:
Indeed! The CI quality improved a lot